### ISSUE TITLE
Footnotes fail the list item label and list item body tag tests for accessibility in digital PDF

### ISSUE THREAD

**Issue by mcgratay** [pressbooks/pressbooks-book#664]
### Description

The PDF accessibility guidelines set out in the Adobe Acrobat User Guide state that 
 
> Lists must have the following structure: A List element must contain List Item Elements. And, List Item Elements can only contain Label Elements and List Item Body Elements. When this rule check fails, the structure of this list is incorrect.

While created via the visual editor tools for ordered and unordered lists do pass this requirement, lists created automatically via the Footnotes feature do not. The Adobe accessibility checker flags footnotes as LBL and LBody errors. 

![Screen Shot 2020-02-05 at 16 36 49](https://user-images.githubusercontent.com/39064028/73969024-9fde9b00-48e8-11ea-8a34-8d867ac16722.png)

View the Adobe User Guide: https://helpx.adobe.com/acrobat/using/create-verify-pdf-accessibility.html?trackingid=KACNN#LblLBody

### Steps to Reproduce

1. Download Adobe Acrobat Pro
2. Create a new chapter in your test book in Pressbooks
3. Add dummy text, then insert a footnote somewhere in the text and save
4. Export to digital PDF 
5. Open in Adobe, then activate the accessibility checker to view whether the footnote passes or fails accessibility requirements 

**Expected behavior:** Digital PDFs meet all known accessibility requirements

**Actual behavior:** Footnotes currently fail accessibility checks in the PDF format.

**Comment by SteelWagstaff** [comment]
Our footnote routine produces `<li>` elements that contain footnote content and `a` elements: https://github.com/pressbooks/pressbooks/blob/92bb307be76a5ee20a88e246459cac4b30403fce/inc/shortcodes/footnotes/class-footnotes.php#L147. Apparently, creating accessible lists requires different tag structure for PDFs than are expected for HTML. 

> The PDF specification defines list structure in section 14.8.4.3.3 (List Elements). The structure types for lists in PDF documents are:
> L - the List tag, which contains one or more LI tags.
> LI - the List Item tag. List item tags can contain Lbl and LBody tags.
> Lbl - the list item label. Contains distinguishing information such as a item number or bullet character.
> LBody - the list item body. Contains list item content, or in the case of a nested list, it may contain additional List tag trees.
Source: https://www.w3.org/TR/WCAG20-TECHS/PDF21.html

See also:
![Screenshot from 2020-06-20 18-40-21](https://user-images.githubusercontent.com/13485451/85213552-8aad5a80-b325-11ea-8538-ec4f23b2a409.png)

https://www.w3.org/TR/WCAG20-TECHS/PDF21.html

**Comment by ricardopressbooks** [comment]
Prince creates Footnotes with the following structure:
```
Reference
    Lbl
        [footnote call]
Note
    Lbl
        [footnote marker]
    [footnote body]
```
There isn't enough information about rules for References with a Lbl. 
I wrote to the Prince forums to get more related help:https://www.princexml.com/forum/topic/4454/accessibility-in-footnotes?p=1#22288
They will review the Footnotes structure.

@SteelWagstaff in short term I can see 2 options for this task:
- Wait for a solution from Prince, since this is an issue related to how this tool internally creates the footnotes structure
- Make a refactor and avoid Prince for footnotes. It sounds like a big task, since we should take into account size pages (to send footnotes to the end of the pages), calculate sizes for footnotes, and a lot of other rules that Prince solves natively.

I would recommend wait for a official solution.

**Comment by tw77** [comment]
Interest in a solution for this expressed by Penn State (for PB internal reference: https://pressbooks.zendesk.com/agent/tickets/13978)

**Comment by SteelWagstaff** [comment]
We believe that this is a false positive from the Adobe accessibility checker. See the comments from Prince folks in the forum thread Ricardo opened. We could check back in with them before closing, however.

**Comment by alexdryden** [comment]
We are currently pushing up against this issue as well, and our stakeholders may not permit us to determine that the accessibility checker is issuing a false positive based on the support currently available here and in the Prince forum. In terms of devising a solution that still uses Prince, I was hoping I could float an idea by you for feedback, realizing this is not a solution that you can likely adopt.

I want this to have a very minimal side-effects, so I was thinking about adding processing step in the pdf export routine that would save a temporary version of the xhtml, modify it so that footnotes were not represented as list items (actual modifications pending advice from our stakeholders about reference tagging in PDF), and pass that to Prince. This could be wired up to a PDF export setting for additional control. @SteelWagstaff can you or others advise on this approach, particularly on how sustainable you think it would be with future plans for Pressbooks (e.g., footnote structure, use of xhtml, etc.)?

**Comment by MichSpieler** [comment]
This was brought up again by UT Chattanooga ([Link for PB internal reference](https://pressbooks.zendesk.com/agent/tickets/27361))

> If I set the theme options to display footnotes as endnotes, the header for the notes fails appropriate nesting, and if I set them to display as regular footnotes, the footnotes fail Lbl/LBody.
